### ISSUE TITLE
Http Middleware

### ISSUE THREAD

**Issue by bbrock25** [spartansystems/rn-booster-kit#6]
Do we want to use redux middleware to interact with various API's?

http://redux.js.org/docs/recipes/ReducingBoilerplate.html

I'm not quite sure how much this buys us, but it could be nice to not have to do the `fetch -> then ->  catch` in our action creators.

**Comment by jasontwong** [comment]
Yeah. Another good reason to do this is if we have a REST API vs WebSocket vs GraphQL setup. Each deployment can then choose which middleware to use.
